libdnf/base/goal.cpp

```diff
@@ -1030,6 +1061,74 @@ void Goal::Impl::add_up_down_distrosync_to_goal(
 }
 
+GoalProblem Goal::Impl::add_group_install_to_goal(
+    base::Transaction & transaction,
+    const std::string & spec,
+    const transaction::TransactionItemReason reason,
+    GoalJobSettings & settings) {
+    auto & cfg_main = base->get_config();
+    bool strict = settings.resolve_strict(cfg_main);
+    sack::QueryCmp cmp = settings.ignore_case ? sack::QueryCmp::IGLOB : sack::QueryCmp::GLOB;
+    comps::GroupQuery group_query(base, true);
+    if (settings.group_with_id) {
+        comps::GroupQuery group_query_id(base);
+        group_query_id.filter_groupid(spec, cmp);
+        group_query |= group_query_id;
+    }
+    if (settings.group_with_name) {
+        comps::GroupQuery group_query_name(base);
+        group_query_name.filter_name(spec, cmp);
+        group_query |= group_query_name;
+    }
+    if (group_query.empty()) {
+        auto problem = transaction.p_impl->report_not_found(GoalAction::INSTALL, spec, GoalJobSettings(), strict);
+        if (strict) {
+            return problem;
+        } else {
+            return GoalProblem::NO_PROBLEM;
+        }
+    }
+    auto allowed_package_types = settings.resolve_group_package_types(cfg_main);
+    auto pkg_settings = GoalJobSettings();
+    pkg_settings.with_provides = false;
+    pkg_settings.with_filenames = false;
+    pkg_settings.nevra_forms.push_back(rpm::Nevra::Form::NAME);
+    for (auto group : group_query) {
+        std::vector<libdnf::comps::Package> packages;
+        // TODO(mblaha): filter packages by p.arch attribute when supported by comps
+        for (auto p : group.get_packages()) {
+            if (any(allowed_package_types & p.get_type())) {
+                packages.emplace_back(std::move(p));
+            }
+        }
+        for (const auto & pkg : packages) {
+            // TODO(mblaha): apply pkg.basearchonly when available in comps
+            auto pkg_name = pkg.get_name();
+            auto pkg_condition = pkg.get_condition();
+            if (pkg_condition.empty()) {
+                auto [pkg_problem, pkg_queue] =
+                    add_install_to_goal(transaction, GoalAction::INSTALL_BY_GROUP, pkg_name, pkg_settings);
+                rpm_goal.add_transaction_group_installed(pkg_queue);
+            } else {
+                // check whether condition can even be met
+                rpm::PackageQuery condition_query(base);
+                condition_query.filter_name(std::vector<std::string>{pkg_condition});
+                if (!condition_query.empty()) {
+                    // remember names to identify GROUP reason of conditional packages
+                    rpm::PackageQuery query(base);
+                    query.filter_name(std::vector<std::string>{pkg_name});
+                    if (!query.empty()) {
+                        add_provide_install_to_goal(fmt::format("({} if {})", pkg_name, pkg_condition), pkg_settings);
+                        rpm_goal.add_transaction_group_installed(*query.p_impl);
+                    }
```

Comment:
If query is empty, then there can be a log event about absence of conditional package. But this comment with very low priority.